File: src/main/java/com/quid/kopring/domain/user/loanhistory/UserLoanHistoryRepository.java
package com.quid.kopring.domain.user.loanhistory;

import com.quid.kopring.userLoanHistory.UserLoanHistory;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserLoanHistoryRepository extends JpaRepository<UserLoanHistory, Long> {

    UserLoanHistory findByBookNameAndIsReturn(String bookName, boolean isReturn);

}

File: src/main/kotlin/com/quid/kopring/userLoanHistory/repository/UserLoanHistoryRepository.kt
package com.quid.kopring.userLoanHistory.repository

import com.quid.kopring.userLoanHistory.UserLoanHistory
import com.quid.kopring.userLoanHistory.type.UserLoanStatus

interface UserLoanHistoryRepository {
    fun findByBookNameAndStatus(bookName: String, status: UserLoanStatus): UserLoanHistory?

    fun findByStatus(status: UserLoanStatus) : List<UserLoanHistory>
}